src/Logic/UsersHelper.php

```diff
@@ -67,39 +69,87 @@ public static function get_user( array $data ): WP_User|bool {
 	}
 
 	/**
-	 * Get a username that is not in use from a desired username.
+	 * Confirms that a user login is unused.
 	 *
-	 * If the desired username is in use, a counter will be appended to it until an unused username is found.
+	 * @param string $user_login The user login to check.
+	 * @param int    $exclude_user_id A user ID to exclude from the check.
 	 *
-	 * @param string $desired_username Desired username.
+	 * @return bool
+	 */
+	public static function is_user_login_unused( string $user_login, int $exclude_user_id = 0 ): bool {
+		global $wpdb;
+
+		$prepared_sql = $wpdb->prepare( "SELECT ID FROM $wpdb->users WHERE user_login = %s", $user_login );
+
+		if ( $exclude_user_id ) {
+			// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- Reason: $sql_prepared is a prepared statement.
+			$prepared_sql = $wpdb->prepare( "$prepared_sql AND ID <> %d", $exclude_user_id );
+		}
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching, WordPress.DB.PreparedSQL.NotPrepared
+		$test = null === $wpdb->get_var( $prepared_sql );
+
+		if ( $test && apply_filters( 'newspack_migration_tools_confirm_user_login_unique_across_coauthors_plus_data', false ) ) {
+			do_action( 'newspack_migration_tools_verify_user_login_unique_across_coauthors_plus_data', $user_login, $exclude_user_id );
+		}
+
+		/*
+		 * Sometimes we have old data in the terms table. Perhaps CAP was turned on at some point, and then turned off.
+		 * Perhaps we had them on, and then migrated them to Guest Contributors. Or perhaps they're coming from
+		 * a custom flavor of WordPress where we didn't remove any data for posterity and so there may
+		 * be a clash. Or perhaps we are updating a user's login because we have to, and therefore
+		 * we should confirm that we don't already have some terms data with the same name.
+		 */
+		if ( $test ) {
+			// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+			$test = null === $wpdb->get_var( $wpdb->prepare( "SELECT term_id FROM $wpdb->terms WHERE name = %s", $user_login ) );
+		}
+
+		return $test;
+	}
+
+	/**
+	 * Get a user login that is not in use, starting with a desired user login.
+	 *
+	 * If the desired user login is in use, a counter will be appended to it until an unused user login is found.
+	 *
+	 * @param string $desired_user_login Desired user login.
 	 *
 	 * @return string An unused username.
 	 */
-	public static function get_unused_username( string $desired_username ): string {
-		$original_username = $desired_username;
-		$desired_username  = trim( $desired_username );
-		$max_length        = 60;
-		if ( strlen( $desired_username ) >= $max_length ) {
-			$desired_username = trim( mb_substr( $desired_username, 0, $max_length ) );
+	public static function get_unused_user_login( string $desired_user_login ): string {
```

Comment:
I think I liked the naming with username better. I think only the DB calls it user_login. Keeping username matches it with all other functions having to do with that in WP: https://developer.wordpress.org/?s=username Using 'login' is more of a verb as I see it in WP code.